Okta.Sdk.Abstractions/BaseOktaClient.cs

```diff
@@ -68,6 +73,28 @@ public BaseOktaClient(OktaClientConfiguration apiClientConfiguration = null, Htt
                 userAgentBuilder);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BaseOktaClient"/> class using the specified <see cref="HttpClient"/>.
+        /// </summary>
+        /// <param name="apiClientConfiguration">
+        /// The client configuration. If <c>null</c>, the library will attempt to load
+        /// configuration from an <c>okta.yaml</c> file or environment variables.
+        /// </param>
+        /// <param name="httpClient">The HTTP client to use for requests to the Okta API.</param>
+        /// <param name="logger">The logging interface to use, if any.</param>
+        public BaseOktaClient(
```

Comment:
I'm not actually sure we need a public constructor for this base class.  Technically all instantiation should go through one of the derived classes, so by making the main constructor above `protected` and removing this one we can enforce that.  But I left this in since I don't know for certain if there are any use cases where we need it.

Reply:
Agree. Let's change `BaseOktaClient` constructors to `protected`.

Reply:
Well specifically I will leave the other constructor as protected and remove this one.